repository/src/main/java/org/jbpm/process/workitem/repository/service/RepoService.java

```diff
@@ -121,17 +125,62 @@ protected void loadModules() {
         }
     }
     
+    /*
+     * Service related operations with resultMap
+     */
+
+    public void addService(RepoData service, Map<String, List<String>> resultMap) {
+        List<RepoData> repoDataList = services.stream().filter(repoData -> repoData.getName().equals(service.getName())).collect(Collectors.toList());
+        if (repoDataList.isEmpty() && !services.isEmpty()) {
+            addNewService(service, resultMap);
+        } else {
+            skipService(service, resultMap);
+        }
+
+        listeners.forEach(listener -> listener.onServiceTaskAdded(service));
+    }
+
     /*
      * Service related operations
      */
-    
-    public void addService(RepoData service) {        
+
+    public void addService(RepoData service) {
         services.add(service);
         storage.onAdded(service);
 
         listeners.forEach(listener -> listener.onServiceTaskAdded(service));
     }
 
+    private void addNewService(RepoData service, Map<String, List<String>> resultMap) {
+        List<String> createdList = resultMap.get(CREATED);
+        if (createdList == null) {
+            createdList = new ArrayList<>();
+            resultMap.put(CREATED, createdList);
+        }
+        createdList.add(service.getName());
+        addService(service);
+    }
+
+    public void removeServiceTask(String serviceTaskId, boolean needDisable) {
```

Comment:
@mswiderski Why do we need the needDisable parameter? Does it hurt if this was disabled always?

When is it an applicable situation when we'd like to remove, but not to disable the Service Task? (In the other PR we set true when the Service Task is used by no project.) I don't see why the UI should invoke the service in case of the Service Task still being installed still in a project.

Reply:
@jstastny-cz, the logic in here https://github.com/kiegroup/jbpm-wb/pull/1399/files#diff-dcc43a73cf2fdc739aaef2b7a4c084a6R138. The UI do never do not invoke the ST, it still being installed by projects.

Reply:
I don't see much value in valid to call a disable upon removal as I see removal as permeant operation while disable might be just temporal. So I'd remove that parameter

Reply:
It has three situations.
1 ST is enabled and installed by projects.
  We cannot remove it.
2 ST is enabled but does not installed by projects
  My PR can check it and remove these STs and disable it.
3 ST is disabled and does not installed by projects
  We can remove it directly.
I have three solutions.
1 My current PR, we can remove ST if it doesn't installed by projects. if
it is enabled, will disable it.
2 remove needDisable parameter, if the ST doesn't installed by projects, we remove it directly, don't care about it is enable or disable.
3 remove needDisable parameter, if ST is enabled or installed by projects
we can not remove it in both two situations.just told user, if you want to remove the ST, you need to uninstall it from all projects and disable it.

@mswiderski @jstastny-cz which is better? For my side, the last one more 
reasonable.

Reply:
@bxf12315 I don't think it matters if the service task is disabled or not when we removing it. As long as there are no projects using it you can remove directly a enabled or disabled service task

Reply:
@mswiderski got it, PR updated, please review again.